tools/hidl/hidlcomp.cpp

```diff
@@ -7034,38 +7114,87 @@ void HIDLcompiler::Process()
         write_esp_ng();
         write_esp_ng_cpp();
     }
-    
+
     if (isSCM)
         write_clarion_HRPC_interfaces();
 }
 
 
+void HIDLcompiler::processExecutionProfiling()
+{
+    //
+    // metrics execution profiling requires the memory header. Search the services and/or methods to determine
+    // if enabled in any. If so, write the include statement
+    EspServInfo *si;
+    bool profilingEnabled = false;
+    StrBuffer tmp;
+    for (si=servs; si; si=si->next)
+    {
+        StrBuffer serviceProfilingOptions;
+        bool serviceProfileExecutionEnabled = si->getMetaStringValue(serviceProfilingOptions,"profile_execution");
+
+        //
+        // At the HIDL top level, execution profiling is enable if any service is enabled
+        executionProfilingEnabled |= serviceProfileExecutionEnabled;
```

Comment:
Interesting way of updating the higher level flag.  I wonder, is this faster or clearer than the more explicit:
```
if (!executionProfilingEnabled && serviceProfileExecutionEnabled)
    executionProfilingEnabled = true;
```
?

Reply:
Since this is build time, performance is not as critical, but I think the |= is faster. If you'd like the more explicit version, I can change it.